### ISSUE TITLE
[BUG] Adapt node promotion changes from rancher

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#1920]
**Describe the bug**
Rancher removes the ability to promote agent nodes by labeling CAPI machines (see this [comment](https://github.com/rancher/rancher/issues/36480#issuecomment-1039253499))
We need to adapt to this change.

**To Reproduce**
1. Use the latest rancher image as embedded rancher service.
2. Bootstrap a Harvester cluster with a least 3 nodes, no additional nodes will be promoted.

**Expected behavior**

Two additional nodes should be promoted.

**Support bundle**

**Environment:**
 - Harvester ISO version: master
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): VM

**Additional context**

**Comment by abonillabeeche** [comment]
After the CP is complete and a CP node is removed due to hardware failure, can we remove and replace/add a new CP node?

**Comment by bk201** [comment]
> After the CP is complete and a CP node is removed due to hardware failure, can we remove and replace/add a new CP node?

The backend will try to keep 3 CP nodes if possible. So after a failed CP node is deleted, the backend will promote another one.

**Comment by TachunLin** [comment]
Verified **fixed** on master-75cc3ced-head (03/11). Close this issue.

### Result
Two additional nodes (2 and 3) have be promoted to control-plane role on the 4 nodes harvester cluster. 

```
rancher@harvester-node03-220311:~> sudo -i kubectl get nodes
NAME                      STATUS   ROLES                       AGE     VERSION
harvester-node01-220311   Ready    control-plane,etcd,master   169m    v1.22.7+rke2r1
harvester-node02-220311   Ready    control-plane,etcd,master   138m    v1.22.7+rke2r1
harvester-node03-220311   Ready    control-plane,etcd,master   56m     v1.22.7+rke2r1
harvester-node04-220311   Ready    <none>                      6m31s   v1.22.7+rke2r1
```

Rancher embedded: `v2.6.4-rc8`
```
rancher@harvester-node03-220311:~> sudo -i kubectl get deployments -n cattle-system rancher -o yaml | grep image
        image: rancher/rancher:v2.6.4-rc8
        imagePullPolicy: IfNotPresent
```

### Test Information
* Test Environment: 4 nodes harvester on local kvm machines
* Harvester version: master-75cc3ced-head (03/11)
* Rancher version: v2.6.4-rc8

### Verify Steps
1. Create a 4 nodes harvester cluster
2. Wait for all nodes have settle down correctly
3. ssh to harvester control node with VIP
4. Run command `kubectl get nodes` 
5. Run command `sudo -i kubectl get deployments -n cattle-system rancher -o yaml | grep image`
6. Run k9s
7. Check the rancher pod details
  ![image](https://user-images.githubusercontent.com/29251855/158167816-3caf3434-1716-462d-bfa5-d1340b967c85.png)
